File: sdk/client/javascript/openapi-test/test/personal/PersonalUnlockAccount.test.js
const OpenSdk = require("opensdk-javascript");
const { expect } = require("@jest/globals");
const { RPC } = require("../constant");

const sdk = new OpenSdk(new OpenSdk.ApiClient(RPC));

describe('personal_unlockAccount API', () => {
    test('should return personal_unlockAccount', (done) => {

        let callbackOne = function (error, data, response) {
            expect(error).toBeNull();
            expect(data.jsonrpc).toBe("2.0");
            expect(data.result).toBeDefined()
            done();
        };
        
        const address = "0x6b1ac7bda0073095df1d434c39d66fb31a592bdd";
        const passphrase = "helloWorld";
        const duration = 30;

        sdk.personal.unlockAccount(address, passphrase, {duration}, callbackOne);
    });
});

File: sdk/client/java/openapi-test/src/test/java/opensdk/sdk/apis/personal/PersonalUnlockAccountTest.java
package opensdk.sdk.apis.personal;

import opensdk.sdk.apis.constant.UrlConstants;
import opensdk.sdk.models.PersonalUnlockAccountResponse;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.klaytn.OpenSDK;

import java.io.IOException;

@DisplayName("Personal RPC Test")
public class PersonalUnlockAccountTest {
    private final OpenSDK sdk = new OpenSDK(UrlConstants.SERVER_URL);

    @Test
    @DisplayName("RPC personal_unlockAccount")
    void whenRequestValid_ThenCall200ResponseReturns() throws IOException {
        String address = "0xb1ab1f758e0d6398c568936400ea94825c4ebdc2";
        String passphrase = "helloWorld";
        int duration = 30;

        PersonalUnlockAccountResponse response = sdk.personal.unlockAccount(address, passphrase, duration)
                .send();
        response.getResult();
    }
}
